File: cs50x/week_06/mario-less.py
from cs50 import get_int


def main():
    # prompt the user for input
    height = get_height()
    # print pyramid
    for i in range(1, height + 1):
        for j in range(height, i, -1):
            print(" ", end="")
        for k in range(i):
            print("#", end="")
        print()


def get_height():
    while True:
        n = get_int("How many steps should the pyramid have?: ")
        if n >= 1 and n <= 8:
            break
    return n


main()

File: cs50x/week_01/mario-less.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int n;
  
    // checking for valid input
    do
    {
        // Prompt user for size of the pyramid
        n = get_int("How many steps should the pyramid have?: ");
    }
    while (n < 1 || n > 8);
  
    // For each row
    for (int i = 1; i < n + 1; i++)
    {
        // inverse pyramid of spaces
        for (int j = n; j > i; j--)
        {
            // show blank space
            printf(" ");
        }
      
        // pyramid of hashes
        for (int k = 0; k < i; k++)
        {
            // show hash
            printf("#");
        }
      
        // Move to next row
        printf("\n");
    }
}
